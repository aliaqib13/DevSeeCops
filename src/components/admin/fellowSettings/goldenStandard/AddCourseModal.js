import React, { Component } from 'react';
import { Modal, Button } from 'antd';
import AddCourse from '../../editCourse/add/AddCourse';

export default class AddCourseModal extends Component {
    constructor(props) {
        super(props);
        this.state = {
            confirmLoading: false,
        };
    }

    onClose = () => {
        const { onClose } = this.props;
        onClose();
    }

    render() {
        const {
            visible, courseTypes, addNewCategory, deleteCategory, categories, createCourse,
            uploadCourseImage, uploadFile, updateTags, fetchTags, searchByCourseTags,
        } = this.props;
        const { confirmLoading } = this.state;
        return (
            <Modal
                title="Add Course"
                width={960}
                visible={visible}
                confirmLoading={confirmLoading}
                onCancel={this.onClose}
                destroyOnClose
                footer={[
                    <Button onClick={this.onClose} key="onCancel">Close</Button>,
                ]}
            >
                <AddCourse
                    addNewCategory={addNewCategory}
                    deleteCategory={deleteCategory}
                    categories={categories}
                    createCourse={createCourse}
                    uploadCourseImage={uploadCourseImage}
                    is_template
                    uploadVideo={uploadFile}
                    updateTags={updateTags}
                    fetchTags={fetchTags}
                    searchByCourseTags={searchByCourseTags}
                    courseTypes={courseTypes}
                />
            </Modal>
        );
    }
}
