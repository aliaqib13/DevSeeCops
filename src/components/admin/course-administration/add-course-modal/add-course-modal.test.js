import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddCourseModal from './index';

describe('AddCourseModal', () => {
    let component; let
        props;

    beforeEach(() => {
        props = {
            onClose: jest.fn(),
        };
        component = shallow(<AddCourseModal {...props} />);
    });

    it('Should render successfully', () => {
        const modal = component.find('Modal');
        expect(modal.exists()).toBeTruthy();
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should call onClose method when click on Close button', () => {
        component.props().footer[0].props.onClick();
        expect(props.onClose).toBeCalledTimes(1);
    });

    it('Should render AddCourse component', () => {
        const AddCourseComponent = component.find('Form(withRouter(AddCourse))');
        expect(AddCourseComponent.exists()).toBeTruthy();
    });
});
