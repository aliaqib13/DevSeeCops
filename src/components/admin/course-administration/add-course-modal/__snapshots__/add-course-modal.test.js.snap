// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseModal Should render successfully 1`] = `
<Modal
  confirmLoading={false}
  destroyOnClose={true}
  footer={
    Array [
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
      >
        Close
      </Button>,
    ]
  }
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Add Course"
  transitionName="zoom"
  visible={false}
  width={960}
>
  <Form(withRouter(AddCourse)) />
</Modal>
`;
